src/Language/PureScript/Errors.hs

```diff
@@ -1190,11 +1190,17 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs relPath) e = fl
     printRow f t = markCodeBox $ indent $ f prettyDepth t
 
     -- If both rows are not empty, print them as diffs
+    -- If verbose print all rows else only print unique rows
     printRows :: Type a -> Type a -> (Box.Box, Box.Box)
-    printRows r1@RCons{} r2@RCons{} = let
-      (sorted1, sorted2) = filterRows (rowToList r1) (rowToList r2)
-      in (printRow typeDiffAsBox sorted1, printRow typeDiffAsBox sorted2)
-    printRows r1 r2 = (printRow typeAsBox r1, printRow typeAsBox r2)
+    printRows r1 r2 = case (full, r1, r2) of 
+      (True, _ , _) -> (printRow typeAsBox r1, printRow typeAsBox r2)
+
+      (_, r1'@RCons{}, r2'@RCons{}) -> let 
+        (sorted1, sorted2) = filterRows (rowToList r1') (rowToList r2')
+        in (printRow typeDiffAsBox sorted1, printRow typeDiffAsBox sorted2)
+        
+      (_, _, _) -> (printRow typeAsBox r1, printRow typeAsBox r2)
```

Comment:
Sorry to nitpick but a `case` statement of this size should really have the branch expressions (the bit after `->`) on indented new lines after the patterns.

Also, there's no need to rebind `r1` and `r2` to `r1'` and `r2'` here. Why not:

```haskell
(_, RCons{}, RCons{}) ->
  let (sorted1, sorted2) = filterRows (rowToList r1) (rowToList r2)
  in ...
```

Reply:
Fixed. I was thinking the whole time that I did not want to break anything so I just copied and pasted the old implementation without paying attention to that. :disappointed:

Reply:
No worries! This is what code review is for :)